src/main/java/seedu/workbook/model/internship/UniqueInternshipList.java

```diff
@@ -31,17 +32,29 @@
  */
 public class UniqueInternshipList implements Iterable<Internship> {
 
-    private final ObservableList<Internship> internalList = FXCollections.observableArrayList();
-    private final ObservableList<Internship> internalUnmodifiableList = FXCollections
-            .unmodifiableObservableList(internalList);
+    private final ObservableList<Internship> baseInternalList = FXCollections.observableArrayList();
+    private final ObservableList<Internship> unmodifiableBaseInternalList = FXCollections
+            .unmodifiableObservableList(baseInternalList);
+
+    private ObservableList<Internship> uiFacingInternalList = FXCollections.observableArrayList();
+    private final ObservableList<Internship> unmodifiableUiFacingInternalList = FXCollections
+            .unmodifiableObservableList(uiFacingInternalList);
+
+    // wrapper on baseInternalList, is automatically updated whenever
+    // baseInternalList is updated, and is always in sorted order w.r.t
+    // InternshipComparator
+    private final SortedList<Internship> internalListSortedByDate = new SortedList<>(baseInternalList,
+            new InternshipComparator());
```

Comment:
Nice work here as well by making use of `SortedList` to make your life easier while increasing code cleanliness.